src/SLCore.Listeners/Implementation/ListFilesListener.cs

```diff
@@ -21,17 +21,19 @@
 using System;
 using System.Collections.Generic;
 using System.ComponentModel.Composition;
+using System.IO;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
 using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.SLCore.Core;
 using SonarLint.VisualStudio.SLCore.Listener.Files;
 using SonarLint.VisualStudio.SLCore.Listener.Files.Models;
 using SonarLint.VisualStudio.SLCore.State;
 
 namespace SonarLint.VisualStudio.SLCore.Listeners.Implementation
 {
-    [Export(typeof(IListFilesListener))]
+    [Export(typeof(ISLCoreListener))]
     [PartCreationPolicy(CreationPolicy.Shared)]
     public class ListFilesListener : IListFilesListener
```

Comment:
It is still useful as the contract declaration, IMO

Reply:
It's not used => it's useless and should not be there. Dead code is not a clean code.

Another question is, if `ListFilesListener` should be actually renamed to `SLCoreListener`, because that is what it really implements now

Reply:
> Another question is, if ListFilesListener should be actually renamed to SLCoreListener, because that is what it really implements now

No, ISLCoreListener is an empty marker interface used for exporting the listeners to be registered with rpc.

IListFilesListener, as well as other listener interfaces, are used to declare an external contract. While they're not used directly, they serve as the reminder that the class should not be modified without any reason.

Reply:
In the future, these interfaces may be used for painless mocking in some ITs

Reply:
It's easy to add such a thing in the future in case it will be needed in the future. For now, I can't explain why it is needed => it's confusing dead code

Reply:
[#5408](https://github.com/SonarSource/sonarlint-visualstudio/issues/5408)